### PR TITLE
chore: end pg pool after query (stacked 1)

### PR DESCRIPTION
**Pull request by owlas**
Stacked PR1 for #2650 (original big PR: https://github.com/lightdash/lightdash/pull/5612)

Needed for SSH Tunnels because the pool must be killed before we close an SSH tunnel.

### CODE REVIEW

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 1]
```diff
@@ -122,23 +121,20 @@
...
-        }
+        this.config = config;
     }
 
     async runQuery(sql: string) {
+        let pool: Pool | undefined;
         try {
-            const results = await this.pool.query(sql); // automatically checkouts client and cleans up
+            pool = new Pool(this.config);
+            const results = await pool.query(sql); // automatically checkouts client and cleans up
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:137]: :eyes:

**Review by rephus — COMMENTED** [review]
IMO, if all tests pass including the security one, and e2e  , then I think we can safely merge

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rephus) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:137]: yes, as long as it doesn't count as a vulnerability in our stack, is there a way to flag this in code , like the comments we use on  linters...

[the remark this answers is not in the document]
Comment (owlas) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:137]: It looks like we can't auto-dismiss using comments, but I'll add a comment in the code and dismiss in the UI

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.test.ts [hunk 2]
```diff
@@ -30,9 +32,12 @@ describe('PostgresWarehouseClient', () => {
     });
     it('expect schema with postgres types mapped to dimension types', async () => {
         const warehouse = new PostgresWarehouseClient(credentials);
-        (warehouse.pool.query as jest.Mock).mockImplementationOnce(() => ({
-            fields: queryColumnsMock,
-            rows: columns,
+        (pg.Pool as unknown as jest.Mock).mockImplementationOnce(() => ({
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.test.ts:35]: thank you!!!

**Comment by owlas** [comment]
cli tests are running at last!

**Review by ZeRego — APPROVED** [review]

**Pull request merged**